### PR TITLE
Enabled OS location service on Windows

### PR DESCRIPTION
**Pull request by simonhong**
If OS location service is enabled, LocationProviderWinrt will give
location data. If OS location serivce is disabled,
NetworkLocationProvider will be used.

Resolves https://github.com/brave/brave-browser/issues/10722

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [x] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [x] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
See issue's STR for manual testing
`npm run test brave_browser_tests -- --filter=BraveMainDelegateBrowserTest*`

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: app/brave_main_delegate.cc [hunk 1]
```diff
@@ -191,6 +192,9 @@ bool BraveMainDelegate::BasicStartupComplete(int* exit_code) {
       features::kWebUIDarkMode.name,
       blink::features::kPrefetchPrivacyChanges.name,
       features::kReducedReferrerGranularity.name,
+#if defined(OS_WIN)
+      features::kWinrtGeolocationImplementation.name,
```
Comment (iefremov) [inline app/brave_main_delegate.cc:196]: Looks like it is disabled by default in Chromium, is it safe for us to enable it for the whole userbase?

Comment (simonhong) [reply app/brave_main_delegate.cc:196]: I had a same concern after seeing this comment - https://bugs.chromium.org/p/chromium/issues/detail?id=968883#c33
However, I assume that edge already uses it in stable.

**Comment by simonhong** [comment]
Merged because only unrelated android test was failed. This PR only contains windows change.

File: app/brave_main_delegate.cc (hunk 1 above)
Comment (bsclifton) [reply app/brave_main_delegate.cc:196]: Great call on testing w/ Edge 😄

**Pull request merged**